src/vs/workbench/services/contextmenu/electron-sandbox/contextmenuService.ts

```diff
@@ -252,7 +252,10 @@ class NativeContextMenuService extends Disposable implements IContextMenuService
 			this.telemetryService.publicLog2<WorkbenchActionExecutedEvent, WorkbenchActionExecutedClassification>('workbenchActionExecuted', { id: actionToRun.id, from: 'contextMenu' });
 		}
 
-		const context = delegate.getActionsContext ? delegate.getActionsContext(event) : undefined;
+		// Determine whether the action source is contributed via a proxy (e.g. from an external extension)
+		// Context will need to be serializable (e.g. no recursive structures)
+		const contributedAction = (actionToRun instanceof MenuItemAction) && actionToRun?.item?.source;
```

Comment:
Due to the existing context menus already utilising referenced objects in their commands which cannot be serialised (e.g. due to circular references or bigint types), a mechanism is required to determine if the context menu item has been contributed by an external extension.

The method chosen here relies on a `source` existing for external contributions. I'm not a bug fan of this, is there a better way?

Reply:
We shouldn't try to use two different contexts for one toolbar. That is one thing that has always been an issue with exposing menus to extensions, and I'm not sure whether there's a better way to handle it, I will try to discuss that this week.